typing/typedecl_variance.ml

```diff
@@ -263,60 +291,82 @@ let compute_variance_extension env ~check decl ext rloc =
     {decl with type_params = ext.ext_type_params}
     (ext.ext_args, ext.ext_ret_type)
 
-let compute_variance_decl env ~check decl (required, _ as rloc) =
+let compute_variance_decl env ~check id decl (required, _ as rloc) =
+  let check =
+    if check
+    then Some (Type_declaration (id,decl))
+    else None
+  in
```

Comment:
It is better to change the `~check` argument to the expanded option here, this avoid creating a dependency on `id` even in the case where `check` is false, which in turns means that we don't need to change the type of the `compute` field .